packages/runtime/src/next/api-handler/handlers/proxy-preview-mode.ts

```diff
@@ -85,71 +91,63 @@ async function proxyPreviewModeApiRouteHandler(
     } catch {}
   }
 
-  previewModeProxy.once('proxyReq', proxyReq => {
-    proxyReq.removeHeader('X-Makeswift-Preview-Mode')
-
-    // The following headers are Next.js-specific and are removed to prevent Next.js from
-    // short-circuiting requests, breaking routing.
-    proxyReq.removeHeader('X-Invoke-Path')
-    proxyReq.removeHeader('X-Invoke-Query')
-
-    const url = new URL(proxyReq.path, 'http://n')
-
-    url.searchParams.delete('x-makeswift-preview-mode')
+  const {
+    'X-Makeswift-Preview-Mode': xMakeswiftPreviewMode,
+    'X-Invoke-Path': xInvokePath,
+    'X-Invoke-Query': xInvokeQuery,
+     ...headers
+  } = req.headers
```

Comment:
Maybe to make things clearer we can make a `proxyHeaders` object as an instance of `Headers` and a `proxyUrl` as an instance of `URL` and then use that to apply modifications like removing the `X-Makeswift-Preview-Mode` search param and header so it's clear where we're doing intended business logic vs avoiding Next.js-specific gotchas?